src/main/java/org/mockito/internal/util/Primitives.java

```diff
@@ -60,6 +63,17 @@ public static boolean isAssignableFromWrapper(Class<?> valueClass, Class<?> refe
         return (T) PRIMITIVE_OR_WRAPPER_DEFAULT_VALUES.get(primitiveOrWrapperType);
     }
 
+    /**
+     * Returns the boxed default value for the lambda return type (handling primitive and primitive wrapper types).
+     *
+     * @param consumer the consumer function
+     * @param <T> the function's return type
+     * @return the boxed default value for the lambda return type (handling primitive and primitive wrapper types)
+     */
+    public static <T> T defaultValueForConsumerLambda(Consumer<T> consumer) {
+        Class<?>[] typeArgs = TypeResolver.resolveRawArguments(Consumer.class, consumer.getClass());
```

Comment:
Thanks for your suggestion. I checked it and after some modifications (interfaces are considered after classes) I was able to make it work for an anonymous class. Unfortunately it doesn't work with a lambda (Consumer is "an instance of" `ParameterizedType` which is the only supported type).

I've checked also `GenericMetadataSupport` suggested by @bric3, however, it works also only for anonymous classes.

There is probably a possibility to enhance Mockito's internal mechanisms to support both lambdas and Java 9 - see https://github.com/jhalterman/typetools/pull/35/files .